chat.el

```diff
@@ -127,12 +127,21 @@ If no key is found, error."
 ;;; Asynchronous primitives for streaming data from ChatGPT's API
 
 
-(defun chat--async-raw (messages callback finalize)
+
+(defun remove-bad-messages (messages)
+  "Filters out items in MESSAGES with no content.
+MESSAGES is a list of items, where each item is a list of (key . value) pairs.
+Returns a new list of items where each item has a non-nil 'content' key.
+Makes it still work when user sends nothing in chat."
+  (seq-filter (lambda (item) (not (null (cdr (assoc "content" item))))) messages))
```

Comment:
You don't actually need `(not (null v))`, `v` is sufficient. `not` is actually an alias for `null`, so this is equivalent to `(not (not v))`.

```suggestion
  (seq-filter (lambda (item) (cdr (assoc "content" item))) messages))
```

Reply:
i ended up using this but just removed the function itself